thingsboard_gateway/tb_utility/tb_gateway_remote_configurator.py

```diff
@@ -576,26 +576,40 @@ def _handle_remote_logging_level_update(self, config):
 
     # HANDLERS SUPPORT METHODS -----------------------------------------------------------------------------------------
     def _apply_connection_config(self, config) -> bool:
-        apply_start = time() * 1000
-        old_tb_client = self._gateway.tb_client
+        old_tb_client_config_path = self._gateway.tb_client.get_config_folder_path()
+        old_tb_client_config = self._gateway.tb_client.config
+        connection_logger = getLogger('tb_connection')
         try:
-            old_tb_client.disconnect()
+            self._gateway.tb_client.disconnect()
+            self._gateway.tb_client.stop()
 
-            connection_logger = getLogger('tb_connection')
-            new_tb_client = TBClient(config, old_tb_client.get_config_folder_path(), connection_logger)
+            while self._gateway.tb_client.client.is_connected():
```

Comment:
This loop can cause the gateway stuck, probably, it will be better to add check is the gateway stopped or no, to avoid hold before stop.